File: Javascript/Kyu-6/Duplicate-encoder.js
// Duplicate encoder
function duplicateEncode(word){
    
    word = word.toLowerCase();
    let newStr = '';

    for ( i in word ) {
        if ( (word.split(word[i]).length -1) == 1 ) {
            newStr += '(';
        } else {
            newStr += ')';
        }
        
    };

    return newStr

};

duplicateEncode('din') // --> "((("
duplicateEncode('recede') // --> "()()()"
duplicateEncode('Success') // --> ")())())"

File: Python/6-Kyu/Duplicate-encoder.py
# Duplicate Encoder
def duplicate_encode(word):
    
    return ''.join(['(' if word.lower().count(letter) == 1 else ')' for letter in word.lower()])

# Tests
print(duplicate_encode("din"))  # --> "((("
print(duplicate_encode("recede"))  # --> "()()()"
print(duplicate_encode("Success"))  # --> ")())())","should ignore case"
print(duplicate_encode("(( @"))  # --> "))(("
